scripts/docs/generate_asset_documentation.py

```diff
@@ -36,27 +39,110 @@ def create_asset_doc(spec_file_name, asset_type):
     with open(full_asset_file_name, 'w') as f:
         f.write(str(doc))
 
-    return asset_type, asset["name"], asset_file_name
+    return asset_type, asset["name"], asset_file_name, asset.get("description", "")
+
+
+def create_environment_doc(asset, asset_config):
+    """Generate environment document content."""
+    doc = snakemd.new_doc()
+
+    doc = add_intro(doc, asset)
+    link = "https://ml.azure.com/registries/azureml/environments/{}/version/{}".format(asset["name"], asset["version"])
+    doc.add_paragraph("**View in Studio**:  [{}]({})".format(link, link))
+    # doc.add_paragraph("**View in Studio**:  <a href=\"{}\" target=\"_blank\">{}</a>".format(link, link))
+    doc = add_mcr_image(doc, asset, asset_config)
+    doc = add_docker_context(doc, asset, asset_config)
+    doc = add_additional_details(doc, asset)
+
+    description = None
+    name = None
+    categories = None
+    return doc, description, name, categories
 
 
-def add_intro(doc, asset):
+def create_component_doc(asset, asset_config):
+    """Generate component document content."""
+    doc = snakemd.new_doc()
+
+    doc = add_intro(doc, asset)
+    doc = add_inputs(doc, asset)
+    doc = add_outputs(doc, asset)
+    doc = add_additional_details(doc, asset)
+
+    description = None
+    name = None
+    categories = None
+    return doc, description, name, categories
+
+
+def create_model_doc(asset, asset_config):
+    """Generate model document content."""
+    doc = snakemd.new_doc()
+
+    doc = add_intro(doc, asset)
+
+    doc = add_inputs(doc, asset)
+    doc = add_outputs(doc, asset)
+    doc = add_additional_details(doc, asset)
+
+    description = None
+    name = None
+    categories = None
+    return doc, description, name, categories
+
+
+def add_mcr_image(doc, asset, asset_config):
+    """Add MCR Image."""
+    img = ":".join([str(asset_config.extra_config_as_object().get_full_image_name()), asset["version"]])
+    doc.add_paragraph("**Docker image**: " + img)
+    return doc
+
+
+def add_docker_context(doc, asset, asset_config):
+    """Add docker context content."""
+    doc.add_heading("Docker build context", level=2)
+    context = {}
+    dockerfile = ""
+    for context_file in search(str(asset_config.extra_config_as_object().context_dir_with_path) + "/*"):
+        content = ""
+        with open(context_file, "r") as f:
+            # do not expect nested structure for now
+            content = f.read()
+        filename = os.path.basename(context_file)
```

Comment:
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to should help here.

Reply:
will have that as a follow up!